app/components/message_attachments/action_button/action_button.js

```diff
@@ -22,9 +23,15 @@ export default class ActionButton extends PureComponent {
 
     handleActionPress = preventDoubleTap(() => {
         const {actions, id, postId} = this.props;
+        this.handleDisableButtonState();
```

Comment:
Perhaps you can do `this.setState({isDisabled: true}, this.handleDisableButtonState);`

and then in handleDisableButtonState just call the timeout, this way you ensure that the timeout is being called after the state was set

Reply:
Thanks